client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -129,13 +128,9 @@ const Header: React.FunctionComponent = () => {
 
 	React.useEffect( () => {
 		if ( newUser && newUser.bearerToken && newUser.username && ! newSite ) {
```

Comment:
Would it help to add a `! isCreatingSite` check to this `if` statement, like we're using in the `OnboardingEdit` component? Ref: https://github.com/automattic/wp-calypso/blob/fbd2194d6426b84f607a67ffd5e6bc9646820d35/client/landing/gutenboarding/onboarding-block/edit.tsx#L38 and https://github.com/automattic/wp-calypso/blob/fbd2194d6426b84f607a67ffd5e6bc9646820d35/client/landing/gutenboarding/onboarding-block/edit.tsx#L85

Reply:
I managed to track down what was re-running the side-effect 607c9c6

Curse these `useEffect()`s :shake-fist:

Reply:
Ah, gotcha. Nice one!
Yes, it's the one thing that really frustrates me about using hooks over class component lifecycle methods, it always seemed like it was going to be simpler to reason about, but then we have situations like this one! 😄